cirq-core/cirq/ops/classically_controlled_operation_test.py

```diff
@@ -385,7 +385,7 @@ def test_condition_removal():
     op = op.without_classical_controls()
     assert not cirq.control_keys(op)
     assert not op.classical_controls
-    assert set(map(str, op.tags)) == {'t1'}
+    assert not op.tags
```

Comment:
@95-martin-orion This changes the assumption that tags are preserved when calling `with_classical_controls`. This looks reasonable to me; since even in the original version, only `t1` was preserved instead of both `t1` and `t2`. 

Just want to make sure this looks good to you.

Reply:
The docstrings for with/without_classical_controls both describe how tags are handled. If that behavior needs to change then the docstrings need updated to be consistent. (Probably can just change "hide" to "remove" in `with`, and I guess you can delete the blurb in `without` completely since it's no longer applicable).

Reply:
+1 to Dax's comment, other than that I'm on board. Might want a BREAKING_CHANGE label to keep track of this, but I see it as a bugfix - the intent is for _any_ non-tagging operation to eliminate tags.

Reply:
Updated the docstring.